pkg/generate/app/cmd/newapp.go

```diff
@@ -313,6 +313,15 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 			switch {
 			case refInput.ExpectToBuild:
 				glog.V(4).Infof("will add %q secrets into a build for a source build of %q", strings.Join(c.Secrets, ","), refInput.Uses)
+
+				// NOTE: This only works so long as the user can only specify
+				// --build-args from new-build, which will only have one refInput.
+				// If the user becomes able to specify more than one refInput, this
+				// needs to be modified to handle that.
+				if len(c.BuildArgs) > 0 && refInput.Uses.GetStrategy() != generate.StrategyDocker {
+					return nil, fmt.Errorf("Cannot use build arguments, not a Docker build")
+				}
```

Comment:
why not just iterate the objects at the end?  if build-args was passed, and at least one of the objects being created is not a buildconfig w/ a docker strategy, throw an error.